actionpack/lib/action_dispatch/http/permissions_policy.rb

```diff
@@ -145,68 +298,28 @@ def initialize_copy(other)
         MSG
 
         if sources.first
-          @directives[directive] = apply_mappings(sources)
+          validate_sources(sources)
+          @directives[directive] = sources
         else
           @directives.delete(directive)
         end
       end
     end
 
-    def build(context = nil)
```

Comment:
This is a breaking change since this method and the entire object is part of the public API. I think instead of doing what you did with two different helper modules, we should be accepting the builder as optional argument of this method and use that object to build the directives.